core/templates/dev/head/domain/topic_viewer/ReadOnlyTopicObjectFactory.ts

```diff
@@ -0,0 +1,128 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating instances of frontend topic-data domain
+ * objects.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+
+import { Injectable } from '@angular/core';
+
+import { Subtopic, SubtopicObjectFactory } from
+  'domain/topic/SubtopicObjectFactory';
+
+import { SkillSummary, SkillSummaryObjectFactory } from
+  'domain/skill/SkillSummaryObjectFactory';
+
+export class ReadOnlyTopic {
+  _topicName: string;
+  _topicId: string;
+  _canonicalStories: any;
+  _additionalStories: any;
+  _uncategorizedSkills: Array<SkillSummary>;
+  _subtopics: Array<Subtopic>;
+  _degreesOfMastery: any;
+  _skillDescriptions: any;
+
+  constructor(topicName: string, topicId: string, canonicalStories: any,
+      additionalStories: any, uncategorizedSkills: Array<SkillSummary>,
+      subtopics: Array<Subtopic>, degreesOfMastery: any, skillDescriptions) {
+    this._topicName = topicName;
+    this._topicId = topicId;
+    this._canonicalStories = canonicalStories;
+    this._additionalStories = additionalStories;
+    this._uncategorizedSkills = uncategorizedSkills;
+    this._subtopics = subtopics;
+    this._degreesOfMastery = degreesOfMastery;
+    this._skillDescriptions = skillDescriptions;
+  }
+
+  getTopicName(): string {
+    return this._topicName;
+  }
+
+  getTopicId(): string {
+    return this._topicId;
+  }
+
+  // Return type is any as there is no usage
+  getCanonicalStories(): any {
+    return this._canonicalStories.slice();
+  }
+
+  // Return type is any as there is no usage
+  getAdditionalStories(): any {
+    return this._additionalStories.slice();
+  }
+
+  getUncategorizedSkills(): Array<SkillSummary> {
+    return this._uncategorizedSkills.slice();
+  }
+
+  getSubtopics(): Array<Subtopic> {
+    return this._subtopics.slice();
+  }
+
+  // Return type any as no appropriate class found for object construction
```

Comment:
If that's the reason then please create an issue for creating a class for this kinda object or create a new class for the same. 

I tried looking into the backend and I think this can be represented as dict(string: number or None). Can you please check how we can represent this in typescript?

Reply:
https://stackoverflow.com/questions/42211175/typescript-hashmap-dictionary-interface
Maybe this would be helpful^
Basically, if we are sure that the degreesOfMastery is going to be a dict with key skillId: string and value degree: number | null, then let's not use `any`. Maybe we can define an interface and use it?

Reply:
Thank You so much for suggesting this, I will try this.